marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/entitySearch/getMatchingPropertyValues.sjs

```diff
@@ -68,20 +69,29 @@ if(referenceType === 'element') {
 } else if(referenceType === 'field') {
   query = cts.fieldReference(propertyPath);
 } else if(referenceType === 'collection') {
-  query = cts.collectionReference();
+  const collectionsToIgnore = Artifacts.getAllArtifactCollections();
+  collectionsToIgnore.push("http://marklogic.com/entity-services/models", "hub-core-artifact");
+
+  const allCollectionsQuery = cts.collectionQuery(cts.collections());
+  const ignoredCollectionsQuery = queryObj.ignoreArtifactCollections ? cts.collectionQuery(collectionsToIgnore) : cts.collectionQuery([]);
+  query = cts.andNotQuery(allCollectionsQuery, ignoredCollectionsQuery);
 } else {
   const result = lib.buildPathReferenceParts(entityTypeId, propertyPath);
   query = cts.pathReference(result.pathExpression, null, result.namespaces);
 }
 
-var facetValues = cts.valueMatch(query, pattern + "*",
-    ["item-order", "ascending", "limit=" + limit]).toArray().map(String);
-
-if (facetValues.length < limit) {
-  var moreFacetValues = cts.valueMatch(query, "?*" + pattern + "*",
-      ["item-order", "ascending", "limit=" + limit]).toArray().map(String);
-  facetValues = Array.from(
-      [...new Set([...facetValues, ...moreFacetValues])]).slice(0, limit);
+let facetValues = [];
+if(referenceType === 'collection') {
```

Comment:
For better readability, put each block of code here into a separate function - e.g. findMatchingCollections and findMatchingIndexValues. Looks like there's some duplication between the two blocks of code to resolve as well.